solr/core/src/test/org/apache/solr/handler/TestContainerPlugin.java

```diff
@@ -312,6 +340,45 @@ public void testApiFromPackage() throws Exception {
     }
   }
 
+  public static class CC1 extends CC {
+
+  }
+  public static class CC2 extends CC1 {
+
+  }
+  public static class CC implements ConfigurablePlugin<CConfig> {
+    private CConfig cfg;
+
+
+
+    @Override
+    public void initConfig(CConfig cfg) {
+      this.cfg = cfg;
+
+    }
+
+    @EndPoint(method = GET,
+        path = "/hello/plugin",
+        permission = PermissionNameProvider.Name.READ_PERM)
+    public void m2(SolrQueryRequest req, SolrQueryResponse rsp) {
+      rsp.add("config", cfg);
+    }
+
+  }
+
+  public static class CConfig extends PluginMeta {
```

Comment:
This example may be confusing because in general case configuration classes don't have to subclass `PluginMeta`. I propose removing this subclassing here to make it clear that's the case.